src/libs/EmojiUtils.ts

```diff
@@ -477,109 +452,86 @@ const getPreferredEmojiCode = (emoji, preferredSkinTone) => {
  * array of emoji codes, that represents all used variations of the
  * emoji, sorted by the reaction timestamp.
- * @param {Object} emojiAsset
- * @param {String} emojiAsset.name
- * @param {String} emojiAsset.code
- * @param {String[]} [emojiAsset.types]
- * @param {Array} users
- * @return {string[]}
- * */
-const getUniqueEmojiCodes = (emojiAsset, users) => {
-    const emojiCodes = _.reduce(
-        users,
-        (result, userSkinTones) => {
-            _.each(lodashGet(userSkinTones, 'skinTones'), (createdAt, skinTone) => {
-                const emojiCode = getPreferredEmojiCode(emojiAsset, skinTone);
-                if (!!emojiCode && (!result[emojiCode] || createdAt < result[emojiCode])) {
-                    // eslint-disable-next-line no-param-reassign
-                    result[emojiCode] = createdAt;
-                }
-            });
-            return result;
-        },
-        {},
-    );
+ */
+const getUniqueEmojiCodes = (emojiAsset: Emoji, users: TimestampedUsersReactions): string[] => {
+    const emojiCodes: Record<string, string> = Object.values(users ?? {}).reduce((result: Record<string, string>, userSkinTones) => {
+        Object.keys(userSkinTones.skinTones).forEach((skinTone) => {
+            const createdAt = userSkinTones.skinTones[Number(skinTone)];
+            const emojiCode = getPreferredEmojiCode(emojiAsset, Number(skinTone));
+            if (!!emojiCode && (!result[emojiCode] || createdAt < result[emojiCode])) {
+                // eslint-disable-next-line no-param-reassign
+                result[emojiCode] = createdAt;
+            }
+        });
+        return result;
+    }, {});
 
-    return _.chain(emojiCodes)
-        .pairs()
-        .sortBy((entry) => new Date(entry[1])) // Sort by values (timestamps)
-        .map((entry) => entry[0]) // Extract keys (emoji codes)
-        .value();
+    return Object.keys(emojiCodes).sort((a, b) => (new Date(emojiCodes[a]) > new Date(emojiCodes[b]) ? 1 : -1));
 };
 
 /**
  * Given an emoji reaction object and its name, it populates it with the oldest reaction timestamps.
- * @param {Object} emoji
- * @param {String} emojiName
- * @returns {Object}
  */
-const enrichEmojiReactionWithTimestamps = (emoji, emojiName) => {
-    let oldestEmojiTimestamp = null;
+const enrichEmojiReactionWithTimestamps = (emoji: UsersReactionsList, emojiName: string): EnrichedUserReactions => {
+    let oldestEmojiTimestamp: string | null = null;
 
-    const usersWithTimestamps = _.chain(emoji.users)
-        .pick(_.identity)
-        .mapObject((user, id) => {
-            const oldestUserTimestamp = lodashMin(_.values(user.skinTones));
+    const usersWithTimestamps: Record<string, UserReactionsWithTimestamps> = {};
+    Object.keys(emoji.users ?? {}).forEach((id) => {
+        const user = emoji?.users?.[id];
+        const oldestUserTimestamp = Object.values(user?.skinTones ?? {}).reduce((min, curr) => (curr < min ? curr : min));
```

Comment:
initial value is missing in reduce 2nd param. This leads to crash

Reply:
I've added the initial value, can you tell me what have you done to crash the app?

Reply:
I randomly tested emojis on other platform. And then opened that report in android app

<img width="789" alt="Screenshot 2023-11-07 at 5 32 35 PM" src="https://github.com/Expensify/App/assets/108292595/b5cc8189-6858-4fe4-8234-a08ec3c2108c">

https://github.com/Expensify/App/assets/108292595/96894bc6-6417-43ee-aa3d-5e3f1aaa14bd